src/components/Editor/index.js

```diff
@@ -16,44 +17,199 @@ export default class Editor extends Component {
+        this.textAreaChanged = this.textAreaChanged.bind(this);
+    }
+
+    textAreaChanged(text) {
+        this.setState({textAreaValue : text});
+    }
+
+    componentDidMount() {
+        $('#undoDelete').hide();
+    }
+
+    discard() {
+        if (confirm("Any work on this problem will NOT be saved")) {
+            this.props.history.goBack()
+            googleAnalytics('Discard');
+        }
+    }
+
+    saveSolution() {
+        googleAnalytics('Save');
+        this.props.history.goBack()
+    }
+
+    editStep(stepNumber) {
+        // nothing to do if there are no steps
+        let mathStep = $($("#MathHistory").children()[stepNumber]);
+    
+        if (this.state.updateMathFieldMode === false) {
+            $('<div/>', {
+                id: "latestMathStepData",
+            }).hide().appendTo('#MathHistory');
+            $('#latestMathStepData').attr('data-equation', this.state.theActiveMathField.latex());
+            $('#latestMathStepData').attr('data-annotation', this.state.textAreaValue);
+            //latestMathStepData.data('scratch-pad', ScratchPadPainterro.imageSaver.asDataURL());
+        }
+        let updatedMathField = this.state.theActiveMathField;
+        updatedMathField.latex(mathStep.data('equation'));
+        this.setState({
+            theActiveMathField: updatedMathField,
+            textAreaValue: mathStep.data('annotation'),
+            editing: true,
+            updateMathFieldMode : true},
+            this.moveEditorBelowSpecificStep(stepNumber)
+        );
+
+        $('#updateStep').unbind();
+        $('#updateStep').click(() =>
+            this.updateStep(stepNumber)
+        );
+        
+        var content = this.setScratchPadContentData(stepNumber);
+        if (content) {
+            this.applyScratchPadContent(content);
+        } else {
+            this.clearScrachPad();
+        }
+    }
+
+
+    //TODO: lift sratchpadpainterro up to here from my workeditorbuttons so we can use it
+    clearScrachPad() {
+       // ScratchPadPainterro.clearBackground();
+       // ScratchPadPainterro.worklog.current = null;
+        // it is because Painterro displays a modal if we want to replace an existing ScratchPad content
+       // ScratchPadPainterro.worklog.clean = true;
+    }
+
+    setScratchPadContentData(stepNumber, newContent) {
+       // let mathStep = $('.mathStep:eq('+ (stepNumber - 1) +')');
+       // mathStep.data('scratch-pad', newContent);
+    }
+    
+    getScratchPadContentData(stepNumber) {
+        //let mathStep = $('.mathStep:eq('+ (stepNumber - 1) +')');
+        //return mathStep.data('scratch-pad');
+    }
+    
+    applyScratchPadContent(content) {
+       // ClearScrachPad();
+        //ScratchPadPainterro.show(content);
+    }
+
+    updateStep(index) {
+        if (this.state.textAreaValue === '') {
+            NotificationManager.warning('Please provide a description of your work.', 'Warning');
```

Comment:
please use our createAlert method - it wraps NotificationManager, so the notification container can gain focus.